### DISCUSSION TITLE
Framework read api is throwing gateway timeout

### DISCUSSION THREAD

**Discussion by amitpriyadarshi** [Sunbird-Knowlg/Community#186] [Bugs]
@pallakartheekreddy as per our discussion framework read api throws gatway timeout exception. But this was not a consistent issue. We were facing 6 out of 10 times (apx 60%). 
As per the log analysis, issue was because of out of memory exception in taxonomy service pod. Initially the heap memory was configured as 600 MB. Post increasing the memory to 1 GB, issue is resolved (upto certain level). Now the frequency of issue is reduced to 5%.
Please suggest if we can increase the configuration more or what should be the ideal memory configuration value for taxonomy service.

cc: @vinukumar-vs

**Comment by pallakartheekreddy** [comment]
@amitpriyadarshi Earlier, these framework APIs were integrated into the learning service. However, in releases 5.6 and 5.7, we relocated these APIs to the taxonomy service. Consequently, we are encountering a memory exception in the taxonomy service due to an attempt to retrieve a substantial amount of data (15 MB) as a response from the framework read API. 

Therefore, kindly adjust the memory value for the taxonomy service to 2 GB.
@maheshkumargangula

**Reply by pallakartheekreddy** [reply]
@amitpriyadarshi Hope your query is now resolved. Please mark the thread as answered
